mockk/src/main/kotlin/io/mockk/impl/annotations/JvmMockInitializer.kt

```diff
@@ -67,9 +78,113 @@ class JvmMockInitializer(val gateway: MockKGateway) : MockKGateway.MockInitializ
         }
     }
 
+    private fun resolveInstance(target: Any, property: KProperty1<Any, Any>): Any? {
+
+        var obj: Any?
+        val objCls = property.returnType.classifier as? KClass<*>
+
+        if (objCls != null) {
+            obj = objCls.objectInstance
+
+            if (obj == null) {
+                if (objCls.constructors.isEmpty() && objCls.primaryConstructor == null) {
```

Comment:
Isn't `objCls.primaryConstructor == null` always true if `objCls.constructors.isEmpty()` is true?